sdk/keyvault/Azure.Security.KeyVault.Keys/src/DeleteKeyOperation.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Security.KeyVault.Keys
+{
+    /// <summary>
+    /// A long-running operation for <see cref="KeyClient.StartDeleteKey(string, CancellationToken)"/> or <see cref="KeyClient.StartDeleteKeyAsync(string, CancellationToken)"/>.
+    /// </summary>
+    public class DeleteKeyOperation : Operation<DeletedKey>
+    {
+        private readonly KeyVaultPipeline _pipeline;
+        private readonly DeletedKey _value;
+        private Response _response;
+        private bool _completed;
+
+        internal DeleteKeyOperation(KeyVaultPipeline pipeline, Response<DeletedKey> response)
```

Comment:
needs to have a public constructor that takes Id and client to be able to resume the operation.

Reply:
It's a good suggestion, but as we discussed offline with @KrzysztofCwalina, because it's not a server-supported LRO the ID we poll on could be used improperly. Imagine this scenario:

1. User deletes a key and gets an LRO but doesn't wait.
2. Shortly thereafter, they (or someone else) recovers the key.
3. User creates the LRO to monitor the ID from step 1.

In this case, they'd wait forever because the deleted key for which we're polling will never exist - not within reason, anyway.